### ISSUE TITLE
Assignees & Labels ( & Reviewers ) Workflow Protocol

### ISSUE THREAD

**Issue by EkoJR** [semperfiwebdesign/all-in-one-seo-pack#2155]
My first thoughts is to have a workflow (chart) that would help everyone identify where Issues/PR are at, and help in decision making and labeling.

* Issue Created
    * No Status or 'Needs Reproducing'.
* Project Backlog (reproduced and waiting for development)
    * Currently no status.
* Development Stage
    * Assigned and 'In-Development'.
* --- PR is Created ---
* Testing Started
    * Assigned and 'Needs Testing'.
    * Fails repeat Development Stage.
    * Success adds 'Completed Testing'.
* Review Stage
    * Assigned and 'Needs Review'.
    * Request Changes doesn't necessarily repeat Development Stage; unless the concept changed.
    * Success adds 'Completed Review'.
* Review Needs Resolving
    * (Maybe) Assigned and In-Development.
    * Repeat Review Stage.
* Merge Stage
    * Assigned and no status.
* Merge Needs Conflicts Resolved
    * Assigned and 'In-Development' & 'Needs Merge Conflicts Resolved'.
    * Repeat Merge Stage.
* Complete

*If you believe I missed a stage, feel free to edit it here.*

I've added some of the common steps that I know of, but one of the things that seems prominent is confusion on what label to use, as well as confusion with the current status (rare occurrence).

We also seem to use 'In-Development' a lot, which may be interpreted that it's back at the Development Stage. When it technically is not; it could be either at the Testing or Review stage. We also tend to remove 'Needs Testing' or 'Needs Review' when reverting to 'In-Development', and it's currently the only indicator what stage it is in.

There is also a transition from Issues to PR with labels, but I think it's good as is; which there was also a discussion on #1884 . With the last step essentially as 'In-Development' until the final decision is made ( completed or refused for some given reason ).

**Comment by EkoJR** [comment]
I see two separate types of labels.

* 'Needs_&ast;' is an Action requested.
* 'In-&ast;' is a Status/Stage. 

Actions can be a large number of requests needed, and it often switches labels multiple times.

Status would be fewer labels, and works in conjunction with Assignee/Member-Role. Whenever Testing or Review needs Development, then just add 'In-Development' with 'In-Testing' or 'In-Review'.

Label ideas.

* In-Inspect
* In-Development
* In-Testing
* In-Review
* In-Final

**Comment by arnaudbroes** [comment]
In fine with those labels.

There's also a Has Been Reproduced label.

**Comment by wpsmort** [comment]
My feeling is that what we have works with the exception of the stage where  PR has been tested and testing has identified a problem and the PR has been assigned back to the developer to reproduce and fix the problem.  This can also happen at the code review stage.  I feel that this is where we need better Stage labels.

I don't have any suggestions as to what the labels should be but I agree that In Development is not appropriate.  Perhaps labels such as - Failed Testing, Needs Reproducing, Needs Fixing might work?

**Comment by EkoJR** [comment]
@wpsmort Those were my thoughts as well with the Development label, and I noticed it becomes further complex with Needs_&ast; and Completed_&ast; to  fill the gap. My thoughts were to simplify it and allow Needs_&ast; and Completed_&ast; to become secondary.

* Development Stage
    * 'In-Development'.
* --- PR is Created ---
* Testing Started
    * 'In-Testing'.
    * Fails 'In-Testing' & 'In-Development'.
* Review Stage
    * 'In-Review'.
    * Fails 'In-Review' & 'In-Development'.
* Merge Stage
    * 'In-Final'.
    * Fails 'In-Final' & 'In-Development'.
* Complete

Basically, 'In-Development' would be tagged on in conjunction to the stage label it is at, and both comment and roles would further explain what is needed. At least that seems fundamentally sound and simplifies the same method.

**Comment by EkoJR** [comment]
Another thought for `Needs Testing`, `Needs Code Review`, `Completed Testing`, & `Complete Code Review`, could be a `Passed Tests` and 'Passed Review`; which we could use Github's Review feature with. This would be more of a Quality Assessment (QA) approach.
